### PR TITLE
feat(EMS-2640-2586): No PDF - Policy - Other company details - Page setup

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR finalises the page setup for the "Other company details" page in the policy section of a no PDF application.

Note: For future proofing / scalability, have decided to call the field IDs/data the following:
  - `REQUEST_JOINTLY_INSURED_PARTY`
  - `REQUESTED_JOINTLY_INSURED_PARTY`
  - Instead of `NEED_ANOTHER_COMPANY_TO_BE_INSURED` and `OTHER_COMPANY_TO_INSURE`.

Because:
  - The business refers to such "other companies" as a "jointly insured party". The UI/UX refers to this as "another company".
  - A jointly insured party could _not_ be included in the insurance policy.
  - The insurance policy could be rejected because of a requested party.

## Resolution :heavy_check_mark:
- Add new "other company" field IDs, content strings.
- Update E2E test coverage.
- Add `FIELDS` and initial `pageVariables` to the GET controller.
- Create an initial POST controller/endpoint - simply does a redirect (validation will be in another PR).
- Update nunjucks template.

## Miscellaneous :heavy_plus_sign:
- Update `test.yml` to include "another company" and "other company details" E2E directories.

### CODE REVIEW

File: e2e-tests/constants/field-ids/insurance/policy/index.js [hunk 1]
```diff
@@ -49,7 +49,12 @@ export const POLICY = {
   },
   NEED_PRE_CREDIT_PERIOD: 'needPreCreditPeriodCover',
   CREDIT_PERIOD_WITH_BUYER: 'creditPeriodWithBuyer',
-  NEED_ANOTHER_COMPANY_TO_BE_INSURED: 'needAnotherCompanyToBeInsured',
+  REQUEST_JOINTLY_INSURED_PARTY: 'requestJointlyInsuredParty',
+  REQUESTED_JOINTLY_INSURED_PARTY: {
+    COMPANY_NAME: 'companyName',
```
Comment (Zainzzkk) [inline e2e-tests/constants/field-ids/insurance/policy/index.js:54]: Do we need to declare these or can we use the previous `COMPANY_NAME,`

File: src/ui/server/constants/field-ids/insurance/policy/index.ts [hunk 2]
```diff
@@ -49,7 +49,12 @@ export const POLICY = {
   },
   NEED_PRE_CREDIT_PERIOD: 'needPreCreditPeriodCover',
   CREDIT_PERIOD_WITH_BUYER: 'creditPeriodWithBuyer',
-  NEED_ANOTHER_COMPANY_TO_BE_INSURED: 'needAnotherCompanyToBeInsured',
+  REQUEST_JOINTLY_INSURED_PARTY: 'requestJointlyInsuredParty',
+  REQUESTED_JOINTLY_INSURED_PARTY: {
+    COMPANY_NAME: 'companyName',
```
Comment (Zainzzkk) [inline src/ui/server/constants/field-ids/insurance/policy/index.ts:54]: or maybe these should be shared fieldIds?

Comment (ttbarnes) [reply src/ui/server/constants/field-ids/insurance/policy/index.ts:54]: Great idea RE both previous comments, just had a look. Can we tackle this a tech improvement? We can have shared field IDs, but would need to restructure all instances of Companies House to use the same IDs. Don't want to bloat this PR and keen to unblock other tickets

**Review by Zainzzkk — APPROVED** [review]

File: src/ui/server/constants/field-ids/insurance/policy/index.ts (hunk 2 above)
Comment (ttbarnes) [reply src/ui/server/constants/field-ids/insurance/policy/index.ts:54]: Made a note to create a ticket for this

**Pull request merged**